linera-execution/src/execution.rs

```diff
@@ -216,10 +216,8 @@ where
                 contract.execute_message(context, runtime_sender, message)
             }
         });
-        let (runtime_result, call_result) =
-            futures::future::join(runtime_actor.run(), call_result_future).await;
-        let call_result = call_result?;
-        runtime_result?;
+        runtime_actor.run().await?;
+        let call_result = call_result_future.await?;
```

Comment:
I'm assuming that any error in `runtime_actor` will cause a less precise error in `contract.execute_*`.